[DOC] Fix outdated link for K8s naming requirements (#6510)
strimzi/strimzi-kafka-operator: documentation/modules/operators/ref-operator-topic.adoc

If creating a topic in Kafka, use a name that is a valid Kubernetes resource name,
otherwise the Topic Operator will need to create the corresponding `KafkaTopic` with a name that conforms to the Kubernetes rules.

> NOTE: Requirements for identifiers and names in Kubernetes are outlined in link:https://kubernetes.io/docs/concepts/overview/working-with-objects/names[Object Names and IDs^] document.

Reviewer: ```suggestion
NOTE: For more information on the requirements for identifiers and names in Kubernetes, refer to link:https://kubernetes.io/docs/concepts/overview/working-with-objects/names[Object Names and IDs^].
```
